sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSessionCatalogSuite.scala

```diff
@@ -39,23 +40,26 @@ class DataSourceV2SQLSessionCatalogSuite
   override protected def verifyTable(tableName: String, expected: DataFrame): Unit = {
     checkAnswer(spark.table(tableName), expected)
     checkAnswer(sql(s"SELECT * FROM $tableName"), expected)
-    checkAnswer(sql(s"SELECT * FROM default.$tableName"), expected)
     checkAnswer(sql(s"TABLE $tableName"), expected)
   }
 
   override def getTableMetadata(tableName: String): Table = {
     val v2Catalog = spark.sessionState.catalogManager.currentCatalog
     val nameParts = spark.sessionState.sqlParser.parseMultipartIdentifier(tableName)
-    v2Catalog.asInstanceOf[TableCatalog]
-      .loadTable(Identifier.of(Array.empty, nameParts.last))
+    val ident = if (v2Catalog.name == SESSION_CATALOG_NAME) {
+      Identifier.of(nameParts.init.toArray, nameParts.last)
+    } else {
+      Identifier.of(Array.empty, nameParts.last)
```

Comment:
Can't we write `Identifier.of(nameParts.init.toArray, nameParts.last)` for the else branch as well?

Reply:
You are right. Updated.